linera-execution/tests/test_execution.rs

```diff
@@ -936,3 +898,68 @@ async fn test_multiple_messages_from_different_applications() -> anyhow::Result<
 
     Ok(())
 }
+
+/// Tests the system API calls `open_chain` and `chain_ownership`.
+#[tokio::test]
+async fn test_open_chain() {
+    let committee = Committee::make_simple(vec![PublicKey::test_key(0).into()]);
+    let committees = BTreeMap::from([(Epoch::ZERO, committee)]);
+    let ownership = ChainOwnership::single(PublicKey::test_key(1));
+    let child_ownership = ChainOwnership::single(PublicKey::test_key(2));
+    let state = SystemExecutionState {
+        committees: committees.clone(),
+        ownership: ownership.clone(),
+        balance: Amount::from_tokens(5),
+        ..SystemExecutionState::new(Epoch::ZERO, ChainDescription::Root(0), ChainId::root(0))
+    };
+    let mut view = state.into_view().await;
+    let mut applications = register_mock_applications(&mut view, 1).await.unwrap();
+    let (application_id, application) = applications.next().unwrap();
+
+    let context = OperationContext {
+        height: BlockHeight(1),
+        next_message_index: 5,
+        ..make_operation_context()
+    };
+    let message_id = MessageId {
+        chain_id: context.chain_id,
+        height: context.height,
+        index: context.next_message_index,
+    };
+
+    let child_ownership2 = child_ownership.clone();
+    application.expect_call(ExpectedCall::execute_operation(
+        move |runtime, _context, _operation| {
+            assert_eq!(runtime.chain_ownership().unwrap(), ownership);
+            let chain_id = runtime.open_chain(child_ownership2, Amount::ONE).unwrap();
+            assert_eq!(chain_id, ChainId::child(message_id));
+            Ok(RawExecutionOutcome::default())
+        },
+    ));
```

Comment:
Optional suggestion: I just wanted to suggest an alternative style I use for this sort of cloning for closures. Feel free to keep it as is though, I just thought it might be worth mentioning :sweat_smile: 

```suggestion
    application.expect_call(ExpectedCall::execute_operation({
        let child_ownership = child_ownership.clone();
        move |runtime, _context, _operation| {
            assert_eq!(runtime.chain_ownership().unwrap(), ownership);
            let chain_id = runtime.open_chain(child_ownership, Amount::ONE).unwrap();
            assert_eq!(chain_id, ChainId::child(message_id));
            Ok(RawExecutionOutcome::default())
        }
    }));
```

Reply:
You're right, that's much better!